### ISSUE TITLE
traefik + consul acme.tlsChallenge invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by dazoot** [containous/traefik#3830]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I am testing traefik + consul backend and trying to generate a ACME Certificate by using `acme.tlsChallenge` method. We are trying to switch to `consul` from `file` backend. In `file` backend mode all works fine.

### What did you expect to see?

A valid SSL certificate.

### What did you see instead?

No certificate was generated and had `Error in Go routine: runtime error: invalid memory address or nil pointer dereference` in the log.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.0-rc3
Codename:     maroilles
Go version:   go1.10.3
Built:        2018-08-01_01:36:03PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
logLevel = "INFO"
defaultEntryPoints = [ "https" ]

[traefikLog]
  filePath = "/var/log/traefik/traefik.log"

[accessLog]
  filePath = "/var/log/traefik/access.log"
  format = "json"

[accessLog.fields]
  defaultMode = "drop"

[accessLog.fields.names]
  "RequestHost" = "keep"
  "ClientHost" = "keep"
  "RequestLine" = "keep"
  "FrontendName" = "keep"
  "BackendAddr" = "keep"
  "RequestProtocol" = "keep"

[accessLog.fields.headers]
  defaultMode = "drop"

[accessLog.fields.headers.names]
  "User-Agent" = "keep"

[entryPoints]
    #[entryPoints.http]
    #address = ":80"
    #[entryPoints.http.redirect]
    #    entryPoint = "https"

    [entryPoints.https]
    address = ":443"
      [entryPoints.https.tls]

    [entryPoints.traefik]
    address = ":9955"

[acme]
email = "acme@site.ro"
# storageFile = "/etc/traefik/acme.json"
storage = "traefikstatic/acme/account"
caServer = "https://acme-v02.api.letsencrypt.org/directory"
entryPoint = "https"
acmeLogging = true
onHostRule = true
# onDemand = true

[acme.tlsChallenge]
#[acme.httpChallenge]
#entryPoint = "http"

[consul]
endpoint = "127.0.0.1:8500"
watch = true
prefix = "traefikstatic"
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
time="2018-08-28T15:31:42+03:00" level=debug msg="LoadCertificateForDomains [static.site.ro]..."
time="2018-08-28T15:31:42+03:00" level=debug msg="Looking for provided certificate to validate [static.site.ro]..."
time="2018-08-28T15:31:42+03:00" level=debug msg="Domains [\"static.site.ro\"] need ACME certificates generation for domains \"static.site.ro\"."
time="2018-08-28T15:31:42+03:00" level=debug msg="Loading ACME certificates [static.site.ro]..."
time="2018-08-28T15:31:42+03:00" level=info msg="legolog: [INFO] [static.site.ro] acme: Obtaining bundled SAN certificate"
time="2018-08-28T15:31:43+03:00" level=info msg="legolog: [INFO] [static.site.ro] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/a08dUyyaOP2VlHb-.....................9n-uT8pXxEJqHA"
time="2018-08-28T15:31:43+03:00" level=info msg="legolog: [INFO] [static.site.ro] acme: Trying to solve TLS-ALPN-01"
time="2018-08-28T15:31:43+03:00" level=debug msg="Challenge Present static.site.ro"
time="2018-08-28T15:31:44+03:00" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
```

**Comment by ldez** [comment]
Hi, thanks for reporting, could you give the lines after `Error in Go routine:`  with the stack trace.

**Comment by dazoot** [comment]
There is no stack trace printed in the log.

**Comment by nmcfaul** [comment]
I had this same issue - stack trace provided.
```
time="2018-09-26T15:00:30Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
goroutine 205 [running]:
runtime/debug.Stack(0x769e27, 0xc0000b8230, 0x25b8557)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x2155680, 0x442da20)
	/go/src/github.com/containous/traefik/safe/routine.go:148 +0x85
github.com/containous/traefik/safe.GoWithRecover.func1.1(0x26652b8)
	/go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
panic(0x2155680, 0x442da20)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/containous/traefik/acme.(*challengeTLSProvider).Present(0xc000364b40, 0xc0006ab740, 0x13, 0xc000d8f710, 0x2b, 0xc0008d9980, 0x57, 0x0, 0x0)
	/go/src/github.com/containous/traefik/acme/challenge_tls_provider.go:80 +0x17a
github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*tlsALPNChallenge).Solve(0xc000848f60, 0xc00008b960, 0x6a, 0xc0009e2cd0, 0xb, 0xc0009e2cc7, 0x7, 0xc000d8f710, 0x2b, 0x0, ...)
	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/tls_alpn_challenge.go:34 +0x1ad
github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).solveChallengeForAuthz(0xc0001b9980, 0xc000cf8360, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:631 +0x5d6
github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).ObtainCertificate(0xc0001b9980, 0xc000644fb0, 0x1, 0x1, 0x1, 0x0, 0x0, 0x42b900, 0x266a910, 0xc0008fdd68, ...)
	/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:434 +0x2b3
github.com/containous/traefik/acme.(*ACME).getDomainsCertificates(0xc0003906c0, 0xc000644fb0, 0x1, 0x1, 0x1, 0x1, 0x1)
	/go/src/github.com/containous/traefik/acme/acme.go:699 +0x1ce
github.com/containous/traefik/acme.(*ACME).LoadCertificateForDomains.func1()
	/go/src/github.com/containous/traefik/acme/acme.go:551 +0x4af
github.com/containous/traefik/acme.(*ACME).runJobs.func1()
	/go/src/github.com/containous/traefik/acme/acme.go:718 +0x44
github.com/containous/traefik/safe.GoWithRecover.func1(0x26652b8, 0xc0003c4d90)
	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
	/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```

**Comment by ldez** [comment]
@nmcfaul Could you give the Træfik version that you use?

**Comment by ldez** [comment]
@nmcfaul Your logs really helped me find the bug, thanks :+1:

**Comment by traefiker** [comment]
Closed by #3934.
